pkg/controllers/csi/provisioner/controller_test.go

```diff
@@ -301,7 +302,7 @@ func TestOneAgentProvisioner_Reconcile(t *testing.T) { //nolint:revive
 		errorfs := &mkDirAllErrorFs{
 			Fs: afero.NewMemMapFs(),
 		}
-		mockClient := &dtclient.MockDynatraceClient{}
+		mockClient := &mockedclient.Client{}
```

Comment:
IMHO we should create mocked client using NewClient(t) if any expectations are defined (by `On(..)`). Then the library checks if expectations are correct or not

Reply:
You are right, thanks for pointing out